src/coreclr/vm/riscv64/profiler.cpp

```diff
@@ -119,17 +90,77 @@ ProfileArgIterator::ProfileArgIterator(MetaSig* pSig, void* pPlatformSpecificHan
 
 ProfileArgIterator::~ProfileArgIterator()
 {
-    // TODO-RISCV64-CQ: copied codes from loongarch however not yet tested.
-    _ASSERTE(!"RISCV64:NYI");
     LIMITED_METHOD_CONTRACT;
 
     m_handle = nullptr;
 }
 
+LPVOID ProfileArgIterator::CopyStructFromRegisters(const ArgLocDesc* sir)
+{
+    struct Func
+    {
+        static inline const BYTE* postIncrement(const BYTE *&p, int offset)
+        {
+            const BYTE* orig = p;
+            p += offset;
+            return orig;
+        }
+    };
+
+    WRAPPER_NO_CONTRACT;
+    _ASSERTE(m_handle);
+    PROFILE_PLATFORM_SPECIFIC_DATA* pData = reinterpret_cast<PROFILE_PLATFORM_SPECIFIC_DATA*>(m_handle);
+
+    struct { bool isFloat, is8; } fields[] = {
+        { sir->m_structFields & (STRUCT_FLOAT_FIELD_FIRST | STRUCT_FLOAT_FIELD_ONLY_TWO | STRUCT_FLOAT_FIELD_ONLY_ONE),
```

Comment:
In my guess, I think `fields[0]` is for the first one in the field. If then, I think `STRUCT_FLOAT_FIELD_ONLY_ONE` can be set when only second field is `FLOAT` in `GetRiscv64PassStructInRegisterFlags`. (I don't know it can actually happen. I just searched.) Could you check again?
https://github.com/dotnet/runtime/blob/6d3be9e69dc385722851b0c165255cc4828aec7c/src/coreclr/vm/methodtable.cpp#L3783-L3794

Reply:
It passes the `MixedStructFunc` test case from `slowpathcommon.cs` and it takes a struct with second float field only. But I'll check if it ever happens.